extensions/spine/CCSkeletonAnimation.js

```diff
@@ -48,28 +48,85 @@ sp._atlasLoader = {
  */
 sp.ANIMATION_EVENT_TYPE = {
     START: 0,
-    END: 1,
-    COMPLETE: 2,
-    EVENT: 3
+    INTERRUPT: 1,
+    END: 2,
+    DISPOSE: 3,
+    COMPLETE: 4,
+    EVENT: 5
 };
 
-sp.TrackEntryListeners = function(startListener, endListener, completeListener, eventListener){
+sp.TrackEntryListeners = function(startListener, endListener, completeListener, eventListener, interruptListener, disposeListener){
     this.startListener = startListener || null;
     this.endListener = endListener || null;
     this.completeListener = completeListener || null;
     this.eventListener = eventListener || null;
+    this.interruptListener = interruptListener || null;
+    this.disposeListener = disposeListener || null;
+    this.callback = null;
+    this.callbackTarget = null;
+    this.skeletonNode = null;
+
+    this.start = function(trackEntry) {
+        if (this.startListener) {
+            this.startListener(trackEntry);
+        }
+        if (this.callback && this.callbackTarget) {
```

Comment:
Why callbackTarget is obligated ? `this.callback.call(null ...` won't cause any error in JS